pkg/snapstore/abs_snapstore.go

```diff
@@ -33,10 +38,35 @@ const (
 	AzuriteEndpoint = "AZURE_STORAGE_API_ENDPOINT"
 )
 
+// AzureBlockBlobClientI defines the methods that are invoked from the Azure Block Blob API.
+type AzureBlockBlobClientI interface {
+	DownloadStream(ctx context.Context, o *azblob.DownloadStreamOptions) (azblob.DownloadStreamResponse, error)
+	Delete(ctx context.Context, o *azblob.DeleteOptions) (azblob.DeleteResponse, error)
+	CommitBlockList(ctx context.Context, base64BlockIDs []string, options *blockblob.CommitBlockListOptions) (blockblob.CommitBlockListResponse, error)
+	StageBlock(ctx context.Context, base64BlockID string, body io.ReadSeekCloser, options *blockblob.StageBlockOptions) (blockblob.StageBlockResponse, error)
+}
+
+// azureContainerClientI defines the methods required for container operations.
+type azureContainerClientI interface {
+	NewListBlobsFlatPager(o *azcontainer.ListBlobsFlatOptions) *runtime.Pager[azcontainer.ListBlobsFlatResponse]
+	NewBlockBlobClient(blobName string) AzureBlockBlobClientI
+}
+
+// AzureContainerClient embedes a *azcontainer.Client (its methods are overridden in tests).
```

Comment:
```suggestion
// AzureContainerClient embeds a *azcontainer.Client (its methods are overridden in tests).
```

Reply:
Addressed in [9c20bb0](https://github.com/gardener/etcd-backup-restore/pull/751/commits/9c20bb08172b78477510d00ad5f0c9b0c678cbd7).